analytics/pubstack/pubstack_module.go

```diff
@@ -246,20 +226,22 @@ func (p *PubstackModule) updateConfig(config *Configuration) {
 	p.cfg = config
 	p.closeAllEventChannels()
 
-	if p.isFeatureEnable(amp) {
-		p.eventChannels[amp] = eventchannel.NewEventChannel(eventchannel.BuildEndpointSender(p.httpClient, p.cfg.Endpoint, amp), p.buffsCfg.size, p.buffsCfg.count, p.buffsCfg.timeout)
-	}
-	if p.isFeatureEnable(auction) {
-		p.eventChannels[auction] = eventchannel.NewEventChannel(eventchannel.BuildEndpointSender(p.httpClient, p.cfg.Endpoint, auction), p.buffsCfg.size, p.buffsCfg.count, p.buffsCfg.timeout)
-	}
-	if p.isFeatureEnable(cookieSync) {
-		p.eventChannels[cookieSync] = eventchannel.NewEventChannel(eventchannel.BuildEndpointSender(p.httpClient, p.cfg.Endpoint, cookieSync), p.buffsCfg.size, p.buffsCfg.count, p.buffsCfg.timeout)
-	}
-	if p.isFeatureEnable(video) {
-		p.eventChannels[video] = eventchannel.NewEventChannel(eventchannel.BuildEndpointSender(p.httpClient, p.cfg.Endpoint, video), p.buffsCfg.size, p.buffsCfg.count, p.buffsCfg.timeout)
-	}
-	if p.isFeatureEnable(setUID) {
-		p.eventChannels[setUID] = eventchannel.NewEventChannel(eventchannel.BuildEndpointSender(p.httpClient, p.cfg.Endpoint, setUID), p.buffsCfg.size, p.buffsCfg.count, p.buffsCfg.timeout)
+	p.registerChannel(amp)
+	p.registerChannel(auction)
+	p.registerChannel(cookieSync)
+	p.registerChannel(video)
+	p.registerChannel(setUID)
+}
+
+func (p *PubstackModule) isFeatureEnable(feature string) bool {
+	val, ok := p.cfg.Features[feature]
+	return ok && val
```

Comment:
Could you help me understand what this return statement is trying to highlight?

Reply:
That the feature is found in the map and is enabled. This is the original code from Pubstack. I tried to keep my changes as minimal as possible. This is just extracted to a method.